src/app/components/auth/auth.guard.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private authService: AuthService, private router: Router){
+    
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.authService.isLoggedIn$.pipe(
+      tap((isLoggedIn) => {
+        if (!isLoggedIn) {
+          this.router.navigate(['login']);
+        }
+      })
```

Comment:
you can use filter pipe here
```suggestion
    return this.authService.isLoggedIn$.pipe(
      filter((isLoggedIn) => !isLoggedIn),
      tap((isLoggedIn) => this.router.navigate(['login'])
    )
```